aspnetcore/fundamentals/environments.md

```diff
@@ -1,601 +1,580 @@
+## Environment-based `Startup` class and methods
 
-* Execute `net stop was /y` followed by `net start w3svc` from a command prompt.
-* Restart the server.
+Since the release of .NET 6, the ASP.NET Core project templates place startup code for service configuration and request pipeline processing in the `Program` file. Use of a `Startup` class (`Startup.cs`) with [`Configure`](xref:Microsoft.AspNetCore.Hosting.StartupBase.Configure%2A) and [`ConfigureServices`](xref:Microsoft.AspNetCore.Hosting.StartupBase.ConfigureServices%2A) methods was required before the release of .NET 6 but remains supported in ASP.NET Core. You can use the guidance in this section to control code execution by environment using a `Startup` class in any release of ASP.NET Core when a large number of code changes are required for each environment in use by an app.
 
-#### macOS
+Except for the [`Startup` class conventions](#startup-class-conventions) approach, the `Program` class uses the following general approach to identify the `Startup` class, where the `Startup` class is the type for the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderExtensions.UseStartup%2A?displayProperty=nameWithType> method:
 
-Setting the current environment for macOS can be performed in-line when running the app:
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
 
-```bash
-ASPNETCORE_ENVIRONMENT=Staging dotnet run
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder =>
+            {
+                webBuilder.UseStartup<Startup>();
+            });
+}
 ```
 
-Alternatively, set the environment with `export` prior to running the app:
-
-```bash
-export ASPNETCORE_ENVIRONMENT=Staging
+### Inject `IWebHostEnvironment` into the `Startup` class
+
+Inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into the `Startup` constructor to control code execution. This approach is useful when the app requires configuring startup code for only a few environments with minimal code differences per environment.
+
+In the following example, the environment is held in the `_env` field and controls code execution based on the app's environment:
+
+```csharp
+public class Startup
+{
+    private readonly IWebHostEnvironment _env;
+
+    public Startup(IWebHostEnvironment env)
+    {
+        _env = env;
+    }
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        if (_env.IsDevelopment())
+        {
+            ...
+        }
+        else if (_env.IsStaging())
+        {
+            ...
+        }
+        else
+        {
+            ...
+        }
+    }
+
+    public void Configure(IApplicationBuilder app)
+    {
+        if (_env.IsDevelopment())
+        {
+            ...
+        }
+        else
+        {
+            ...
+        }
+
+        ...
+    }
+}
 ```
```

Comment:
The primary constructor version is simpler and more idiomatic with modern C#, so let's just use that and have one sample.

````suggestion
Inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into the `Startup` constructor to access the current environment during startup configuration.
```

Reply:
We don't have to show the primary constructor version. We could just show a normal constructor as our single sample. Thie startup pattern is an older pattern that is eventually going away, so sticking to the older pattern kinda makes sense. But I definitely don't want to show two version of the same sample using different C# features.